persistence/sql/persister_errorx.go

```diff
@@ -55,14 +54,19 @@ func (p *Persister) ReadErrorContainer(ctx context.Context, id uuid.UUID) (_ *er
 	defer otelx.End(span, &err)
 
 	var ec errorx.ErrorContainer
-	if err := p.GetConnection(ctx).Where("id = ? AND nid = ?", id, p.NetworkID(ctx)).First(&ec); err != nil {
-		return nil, sqlcon.HandleError(err)
-	}
-
-	if err := p.GetConnection(ctx).RawQuery(
-		"UPDATE selfservice_errors SET was_seen = true, seen_at = ? WHERE id = ? AND nid = ?",
-		time.Now().UTC(), id, p.NetworkID(ctx)).Exec(); err != nil {
-		return nil, sqlcon.HandleError(err)
+	if err := p.Transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
+		if err := c.Where("id = ? AND nid = ?", id, p.NetworkID(ctx)).First(&ec); err != nil {
+			return sqlcon.HandleError(err)
+		}
+
+		if err := c.RawQuery(
+			"UPDATE selfservice_errors SET was_seen = true, seen_at = ? WHERE id = ? AND nid = ?",
```

Comment:
Ideally we could use a `RETURNING` clause here, but obviously mysql does not support that...